vaadin-map-flow-parent/vaadin-map-flow/src/main/resources/META-INF/resources/frontend/vaadin-map/synchronization/index.js

```diff
@@ -73,12 +73,8 @@ function synchronizeFeature(target, source, context) {
     target = new Feature();
   }
 
-  target.setGeometry(
-    context.synchronize(target.getGeometry(), source.geometry, context)
-  );
-  target.setStyle(
-    context.synchronize(target.getStyle(), source.style, context)
-  );
+  target.setGeometry(context.lookup.get(source.geometry));
```

Comment:
Nested objects are now synchronized separately, so all we need to do here is just get their reference from the lookup.